### PR TITLE
Update Vault version verification integration test

### PR DESCRIPTION
**Pull request by jaymalasinha**
Signed-off-by: Jaymala Sinha <jaymala@hashicorp.com>

### CODE REVIEW

File: enos/modules/get_local_metadata/main.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+resource "enos_local_exec" "get_revision" {
+  inline = ["git rev-parse HEAD"]
+}
+
+output "revision" {
+  value = trimspace(enos_local_exec.get_revision.stdout)
+}
+
```
Comment (ryancragun) [inline enos/modules/get_local_metadata/main.tf:32]: nit: some extra newlines

File: enos/modules/vault_verify_version/templates/verify-cluster-version.sh [hunk 2]
```diff
@@ -13,25 +18,37 @@ fail() {
...
-  <<< $vault_status
-)
-
-if [[ "$result" != "true" ]]; then
-  fail "expected version $binary_version with build_date $build_date, got status $vault_status"
+case "$release" in
+  *+oss) [[ "$(echo $version |awk -F'.' '{print $2}')" -ge 11 ]] && \
+  version_expected="Vault v$version ($sha), built $build_date" || \
+  version_expected="Vault v$version ($sha)" ;;
+	*+ent) ;;
```
Comment (ryancragun) [inline enos/modules/vault_verify_version/templates/verify-cluster-version.sh:40]: this indentation seems a bit off

File: enos/modules/vault_verify_version/templates/verify-cluster-version.sh [hunk 3]
```diff
@@ -13,25 +18,37 @@ fail() {
...
+	*+ent.fips1402) version_expected="$version_expected (cgo)" ;;
+	*+ent.hsm.fips1402) version_expected="$version_expected (cgo)" ;;
+  *) fail "($release) file doesn't match any known license types"
+esac
+
+version_expected_nosha=$(echo "$version_expected" | awk '!($3="")' | sed 's/  / /' | sed -e 's/[[:space:]]*$//')
+version_output=$("$binpath" version)
+
+if [[ "$version_output" == "$version_expected_nosha" ]] || [[ "$version_output" == "$version_expected" ]]; then
+	echo "Version verification succeeded!"
```
Comment (ryancragun) [inline enos/modules/vault_verify_version/templates/verify-cluster-version.sh:51]: nit: inconsistent indentation in this script

**Review by ryancragun — APPROVED** [review]

File: enos/modules/get_local_metadata/main.tf (hunk 1 above)
Comment (jaymalasinha) [reply enos/modules/get_local_metadata/main.tf]: updated

File: enos/enos-variables.hcl [hunk 4]
```diff
@@ -101,6 +101,24 @@ variable "vault_local_build_tags" {
...
 }
 
+variable "vault_build_date" {
+  description = "The build date for Vault artifact"
+  type        = string
+  default     = null
+}
+
+variable "vault_product_version" {
+  description = "The version of vault we are testing"
```
[the remark this answers is not in the document]
Comment (jaymalasinha) [reply enos/enos-variables.hcl]: updated

File: enos/modules/vault_verify_version/templates/verify-cluster-version.sh (hunk 2 above)
Comment (jaymalasinha) [reply enos/modules/vault_verify_version/templates/verify-cluster-version.sh]: updated

File: enos/modules/vault_verify_version/templates/verify-cluster-version.sh (hunk 3 above)
Comment (jaymalasinha) [reply enos/modules/vault_verify_version/templates/verify-cluster-version.sh]: updated

**Pull request merged**